torch/csrc/jit/passes/create_autodiff_subgraphs.cpp

```diff
@@ -28,7 +28,7 @@ void mergeNodes(Graph & g, Symbol group_node_kind, ArrayRef<Node*> nodes) {
     if(value_map.count(v) > 0) {
       return value_map[v];
     }
-    Value * nv = new_graph->addInput();
+    Value * nv = new_graph->addInput()->setType(v->typeOption());
```

Comment:
Maybe we should add an `addInputLike(Value * other_graph)` option to `ir.h`. This pattern appears a lot because we cannot use copyMetadata (though I forget why we cannot) on inputs. This doesn't have to be done here, just a suggestion.

Reply:
Good idea, will add.